src/ParameterDistributions.jl

```diff
@@ -667,4 +694,145 @@ function transform_unconstrained_to_constrained(
     return transf_xarray
 end
 
+# -------------------------------------------------------------------------------------
+# constructor for numerically optimized constrained distributions
+
+function _moment(m::Integer, d::UnivariateDistribution, c::Constraint)
+    # Rough (hopefully fast) 1D numerical integration of constrained distribution expectation
+    # values. Integrate in constrained space, although constraints can give singular behavior 
+    # near bounds.
+    # Intended use is only on bounded integrals; otherwise run into 
+    # https://github.com/JuliaMath/QuadGK.jl/issues/38
+    min = c.unconstrained_to_constrained(minimum(d))
+    max = c.unconstrained_to_constrained(maximum(d))
+    function integrand(x)
+        log_pdf = logpdf(d, c.constrained_to_unconstrained(x))
+        # jacobian always >= 0; use logs to avoid over/underflow
+        return isinf(log_pdf) ? 0.0 : x^m * exp(log(c.c_to_u_jacobian(x)) + log_pdf)
+    end
+    return quadgk(integrand, min, max, order = 9, rtol = 1e-5, atol = 1e-6)[1]
+end
+function _mean_std(μ::FT, σ::FT, c::Constraint) where {FT <: Real}
+    d = Normal(μ, σ)
+    m = [_moment(k, d, c) for k in 1:2]
+    return (m[1], sqrt(m[2] - m[1]^2))
+end
+function _lognormal_mean_std(μ_u::FT, σ_u::FT) where {FT <: Real}
+    # known analytic solution for lognormal distribution
+    return (exp(μ_u + σ_u^2 / 2.0), exp(μ_u + σ_u^2 / 2.0) * sqrt(expm1(σ_u^2)))
+end
+function _inverse_lognormal_mean_std(μ_c::FT, σ_c::FT) where {FT <: Real}
+    # known analytic solution for lognormal distribution
+    return (log(μ_c) - 0.5 * log1p((σ_c / μ_c)^2), sqrt(log1p((σ_c / μ_c)^2)))
+end
+
+"""
+    bounded_gaussian(name::AbstractString, μ_c::FT, σ_c::FT, lower_bound::FT, upper_bound::FT)
+
+Constructor for a 1D ParameterDistribution consisting of a transformed Gaussian, constrained
+to have support on [`lower_bound`, `upper_bound`], with first two moments μ_c and σ_c^2. The 
+moment integrals can't be done in closed form, so we set the parameters of the distribution
+with numerical optimization.
+
+!!! note
+    The intended use case is defining priors set from user expertise for use in inference 
+    with adequate data, so for the sake of performance we only require that the optimization
+    reproduce μ_c, σ_c to a loose tolerance (1e-5). Warnings are logged when the optimization
+    fails.
+
+!!! note
+    The distribution may be bimodal for σ_c large relative to the width of the bound interval.
+    In extreme cases the distribution becomes concentrated at the bound endpoints. We regard
+    this as a feature, not a bug, and do not warn the user when bimodality occurs.
+"""
+function bounded_gaussian(name::AbstractString, μ_c::FT, σ_c::FT, lower_bound::FT, upper_bound::FT) where {FT <: Real}
```

Comment:
Maybe the name should not be `bounded` as it is not always bounded, and maybe it should allude to the fact that it is a constructor for a `ParameterDistribution` object. How about `create_constrained_gaussian_distribution`? Or something along these lines.

Reply:
Is `constrained_gaussian` OK? (i.e. 'create' and 'distribution' being clear from context). climaformat limits us to 120 character lines, and there's no explicit line continuation marker in Julia.